tokio-util/src/codec/length_delimited.rs

```diff
@@ -493,14 +493,6 @@ impl LengthDelimitedCodec {
                 src.get_uint_le(field_len)
             };
 
-            if n > self.builder.max_frame_len as u64 {
-                return Err(io::Error::new(
-                    io::ErrorKind::InvalidData,
-                    LengthDelimitedCodecError { _priv: () },
-                ));
-            }
-
-            // The check above ensures there is no overflow
             let n = n as usize;
```

Comment:
This cast should replaced with a checked one.
```suggestion
            let n = usize::try_from(n).ok();
```
This converts it to an `Option<usize>`. You can use `Option::and_then` to apply the length adjustment.